lib/run_part.c

```diff
@@ -51,68 +52,78 @@ int run_parts(const char *directory, const char *name, const char *action)
 	int n;
 	int execute_result = 0;
 
-	scanlist = scandir(directory, &namelist, 0, alphasort);
+	int dfd = open (directory, O_PATH | O_DIRECTORY | O_CLOEXEC);
+	if (dfd == -1) {
+		fprintf(shadow_logfd, "failed open directory %s: %s\n", directory, strerror(errno));
+		return (1);
+	}
+
+	scanlist = scandir (directory, &namelist, 0, alphasort);
 	if (scanlist<=0) {
+		(void) close (dfd);
```

Comment:
Why `(void)`?

Reply:
Because we are not interested in potential failures from `close(2)`, since the file descriptors are only read from, and some static analyzers flag unused return values from `close(2)`.

Reply:
I prefer leaving this unsilenced.

Otherwise, one may change the code in a way that it should then check for the return value but forget to do so or remove the cast, and it would result in a silent bug.

For silencing static analyzers, I prefer to just pass flags to them to silence something.